unit_test/test_Matrix.cc

```diff
@@ -222,24 +222,15 @@ void test_Matrix_lambda()
     test_assert(A.op() == blas::Op::NoTrans);
     test_assert(A.uplo() == slate::Uplo::General);
 
-    // SLATE doesn't know distribution.
+    // SLATE detects the distribution.
     GridOrder order;
     int myp, myq, myrow, mycol;
     A.gridinfo( &order, &myp, &myq, &myrow, &mycol );
-    if (mpi_size == 1) {
-        test_assert( order == GridOrder::Col );
-        test_assert( myp == 1 );
-        test_assert( myq == 1 );
-        test_assert( myrow == 0 );
-        test_assert( mycol == 0 );
-    }
-    else {
-        test_assert( order == GridOrder::Unknown );
-        test_assert( myp == -1 );
-        test_assert( myq == -1 );
-        test_assert( myrow == -1 );
-        test_assert( mycol == -1 );
-    }
+    test_assert( order != GridOrder::Unknown );
```

Comment:
Can we more strongly assert that `order == GridOrder::ColMajor`, since tileRank was created with `Layout::ColMajor`?

BTW, is there a reason that `process_1d_grid_cyclic` takes `Layout` enum, while `gridinfo` takes `GridOrder` enum?

Reply:
So, the following all give the same distribution function
```C++
slate::func::process_2d_grid( slate::Layout::ColMajor, p, 1 );
slate::func::process_2d_grid( slate::Layout::RowMajor, p, 1 );
slate::func::process_1d_grid( slate::Layout::ColMajor, p );
```
So, asserting `order == GridOrder::ColMajor` would mean that changing the internals of `gridinfo` to an equally correct answer would break the test.

The main reason is that I wrote `process_1d_grid_cyclic` et al. before I started looking at redoing `gridinfo`.  Given that the Matrix constructors take `GridOrder`'s, I think it'd make sense to change `process_1d_grid_cyclic` et al. to use `GridOrder.`

Reply:
Ok, let's change `process_1d_grid_cyclic`. You can do that here or in another PR if you like.